File: Domain/Models/HorarioMedico.cs
using System.ComponentModel.DataAnnotations;

namespace Domain.Models
{
    public class HorarioMedico
    {
        [Key]
        public int IdHorario { get; set; }

        public int IdMedico { get; set; }

        public int IdConsultorio { get; set; }

        public string DiaSemana { get; set; } = string.Empty;

        public TimeSpan HoraInicio { get; set; }

        public TimeSpan HoraFin { get; set; }

        public bool Estado { get; set; }
    }
}

File: Frontend/medical-appointments/src/app/services/horario-medico.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface HorarioMedico {
  idHorario: number;
  idMedico: number;
  idConsultorio: number;
  diaSemana: string;
  horaInicio: string;
  horaFin: string;
  estado: boolean;
}

@Injectable({
  providedIn: 'root'
})
export class HorarioMedicoService {

  private apiUrl = 'https://localhost:7250/HorariosMedicos';

  constructor(private http: HttpClient) {}

  getHorarios(): Observable<HorarioMedico[]> {
    return this.http.get<HorarioMedico[]>(this.apiUrl);
  }
}